kasa/discover.py

```diff
@@ -528,6 +586,20 @@ def _get_device_instance(
             ) from ex
         try:
             discovery_result = DiscoveryResult(**info["result"])
+            if discovery_result.encrypt_info:
+                encryped_key = discovery_result.encrypt_info.key
+                encrypted_data = discovery_result.encrypt_info.data
+
+                key_and_iv = keypair.decrypt_discovery_key(
+                    base64.b64decode(encryped_key.encode())
+                )
+
+                key, iv = key_and_iv[:16], key_and_iv[16:]
+
+                session = AesEncyptionSession(key, iv)
+                decrypted_data = session.decrypt(encrypted_data)
```

Comment:
I would separate the decryption into its own function that gets only called when there's encrypt info, makes it easier to test and keeps the main flow cleaner.

Reply:
Agreed, will do

Reply:
Done and test added